library/LuaApi.cpp

```diff
@@ -2480,6 +2481,65 @@ static const LuaWrapper::FunctionReg dfhack_kitchen_module[] = {
     {NULL, NULL}
 };
 
+/***** XlsxReader module *****/
+
+static const LuaWrapper::FunctionReg dfhack_xlsxreader_module[] = {
+    WRAPM(XlsxReader, open_xlsx_file),
+    WRAPM(XlsxReader, close_xlsx_file),
+    WRAPM(XlsxReader, open_sheet),
+    WRAPM(XlsxReader, close_sheet),
+    {NULL, NULL}
+};
+
+// internal function to factor out handle extraction
+static void * get_xlsxreader_handle(lua_State *L)
+{
+    if (lua_gettop(L) < 1 || lua_isnil(L, 1))
+    {
+        luaL_error(L, "invalid xlsxreader handle");
+    }
+    luaL_checktype(L, 1, LUA_TLIGHTUSERDATA);
+    return lua_touserdata(L, 1);
```

Comment:
well, I would say I *chose* to use light userdata. I actually originally typechecked it as userdata, but the check failed because it was being passe din as light userdata, so I changed the check : p Lua data shuttling is an entirely new area for me, so I'm kinda looking to you to define what "good enough" is here. The current code works, I've verified, when used properly.